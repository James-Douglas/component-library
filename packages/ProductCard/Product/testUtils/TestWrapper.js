import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
