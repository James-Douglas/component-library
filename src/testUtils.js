// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
