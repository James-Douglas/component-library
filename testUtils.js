import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
