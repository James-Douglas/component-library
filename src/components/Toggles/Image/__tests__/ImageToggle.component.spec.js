import React from 'react';
import { render, fireEvent } from '../../../../testUtils';
import ImageToggle, { getImageToggleContent } from '../ImageToggle.component';

describe('getImageToggleContent()', () => {
  const PictureToggleContentContainer = ({
    // eslint-disable-next-line react/prop-types
    src, srcsets, alt, pictureTitle, id, toggleTitle, description,
  }) => (
    <>
      {getImageToggleContent(src, srcsets, alt, pictureTitle, id, toggleTitle, description)}
    </>
  );

  it('renders with props', () => {
    const { getByText, container } = render(
      <PictureToggleContentContainer
        src="/test/test.jpg"
        alt="test alt"
        pictureTitle="test title"
        id="test-picture-toggle"
        toggleTitle="test title"
        description="test description"
      />,
    );
    expect(getByText('test title')).toBeInTheDocument();
    expect(getByText('test description')).toBeInTheDocument();
    const img = container.querySelector('img');
    expect(img).toHaveAttribute('src', '/test/test.jpg');
    expect(img).toHaveAttribute('alt', 'test alt');
    expect(img).toHaveAttribute('title', 'test title');
  });
});

describe('ImageToggle', () => {
  it('renders with props', () => {
    const { container, getByText } = render(
      <ImageToggle
        id="test-picture-toggle"
        title="toggle title"
        description="test description"
        value="test"
        selectedValue="test"
        name="toggles"
        src="/test/test.jpg"
        alt="test alt"
        pictureTitle="test title"
        toggleTitle="test title"
      />,
    );
    expect(container.innerHTML).toMatchSnapshot();
    expect(getByText('toggle title')).toBeInTheDocument();
    expect(getByText('test description')).toBeInTheDocument();
    const input = container.querySelector('input');
    expect(input.value).toEqual('test');
    expect(input.name).toEqual('toggles');
    expect(input.checked).toBeTruthy();
    const img = container.querySelector('img');
    expect(img).toHaveAttribute('src', '/test/test.jpg');
    expect(img).toHaveAttribute('alt', 'test alt');
    expect(img).toHaveAttribute('title', 'test title');
  });

  it('calls handleToggle when a toggle is selected', () => {
    const handleToggle = jest.fn();
    const { container } = render(
      <ImageToggle
        id="test-picture-toggle"
        title="toggle title"
        description="test description"
        value="test"
        name="toggles"
        src="/test/test.jpg"
        alt="test alt"
        pictureTitle="test title"
        toggleTitle="test title"
        handleToggle={handleToggle}
      />,
    );
    const element = container.firstChild;
    fireEvent.click(element);
    expect(handleToggle).toHaveBeenCalled();
    expect(handleToggle.mock.calls[0][0]).toEqual('test');
  });

  it('calls focus and blur handlers', () => {
    const handleFocus = jest.fn();
    const handleBlur = jest.fn();
    const { container } = render(<ImageToggle id="test" value="testt" handleFocus={handleFocus} handleBlur={handleBlur} />);
    const input = container.querySelector('input');
    fireEvent.focus(input);
    expect(handleFocus).toHaveBeenCalled();
    fireEvent.blur(input);
    expect(handleBlur).toHaveBeenCalled();
  });
});
