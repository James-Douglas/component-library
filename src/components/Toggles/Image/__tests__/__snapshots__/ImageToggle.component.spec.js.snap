// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageToggle renders with props 1`] = `"<div class=\\"sc-AxjAm keHfJL\\"><input tabindex=\\"0\\" id=\\"test-picture-toggle\\" type=\\"radio\\" name=\\"toggles\\" class=\\"sc-AxirZ exDnLo\\" value=\\"test\\" checked=\\"\\"><label for=\\"test-picture-toggle\\" class=\\"sc-AxiKw ikDIEa\\"><div class=\\"sc-AxhCb jOcikB\\"><div class=\\"sc-AxheI QCVbg\\"><div class=\\"sc-Axmtr bzdkyF\\"><picture class=\\"sc-AxhUy gOyhKF\\"><img src=\\"/test/test.jpg\\" alt=\\"test alt\\" title=\\"test title\\" class=\\"sc-AxgMl HpKQS\\"></picture></div><div class=\\"sc-AxmLO jSegvA\\"><div class=\\"sc-fzozJi gyIlcr\\">toggle title</div><div class=\\"sc-fzoLsD fOeHdq\\">test description</div></div></div></div></label></div>"`;
